### PR TITLE
Change generated code to use shorthand truthy and no falsy props

### PR DESCRIPTION
**Pull request by jjwilliams42**
Signed-off-by: Josh Williams <williams.jackj@gmail.com>

The default code generated is a bit verbose, explicitly outputting every true / false property on every component, which goes against the typical React pattern.

Before, the code generator would output false and true props with explicit values:

<Button label="Dashboard" plain={true} primary={false} reverse={false} secondary={false} hoverIndicator={false} active={false} icon={<Dashboard />} />

Now, it will output the code in a more optimized format:

<Button label="Dashboard" plain icon={<Dashboard />} />

I understand that maybe you want all properties output, if so then feel free to reject.

Thanks for the awesome framework!

### CODE REVIEW

File: src/design/code.js [hunk 1]
```diff
@@ -151,7 +151,8 @@ export const generateJSX = ({
           return !(
             (typeof value === 'object' && Object.keys(value).length === 0) ||
             value === '' ||
-            value === undefined
+            value === undefined ||
+            value === false
```
Comment (ericsoderberghp) [inline src/design/code.js]: I think this won't work correctly where the value is overriding a default value. For example, `<Clock />` defaults to `run="forward"`. This is turned off by setting `run={false}`.

File: src/design/code.js [hunk 2]
```diff
@@ -175,6 +176,9 @@ export const generateJSX = ({
             }
             return ` ${name}="${value}"`;
           }
+          if (typeof value === 'boolean') {
```
Comment (ericsoderberghp) [inline src/design/code.js]:
```suggestion
          if (typeof value === 'boolean' && value) {
```

File: src/design/code.js (hunk 1 above)
Comment (jjwilliams42) [reply src/design/code.js]: Ah ok that makes sense, shall I remove this part and leave the shorthand part below? Or just reject this whole PR?

Comment (ericsoderberghp) [reply src/design/code.js]: whichever you’d prefer

File: src/design/code.js (hunk 2 above)
Comment (jjwilliams42) [reply src/design/code.js]: Makes sense, fixed, thank you!

**Review by ericsoderberghp — APPROVED** [review]

**Comment by ericsoderberghp** [comment]
Thanks for your contribution!

**Pull request merged**